File: mRemoteNG/Tools/IeBrowserEmulation.vb
﻿Imports System.IO
Imports Microsoft.Win32
Imports mRemoteNG.App.Runtime

Namespace Tools
    Public Class IeBrowserEmulation
        Private Const BrowserEmulationKey As String = "Software\Microsoft\Internet Explorer\Main\FeatureControl\FEATURE_BROWSER_EMULATION"
        Private Shared _previousIeBrowserEmulationValue As Integer = 0
        Public Shared Sub Register()
            Try
                Dim registryKey As RegistryKey = Registry.CurrentUser.OpenSubKey(BrowserEmulationKey, True)
                If registryKey Is Nothing Then
                    Registry.CurrentUser.CreateSubKey(BrowserEmulationKey)
                    registryKey = Registry.CurrentUser.OpenSubKey(BrowserEmulationKey, True)
                    If registryKey Is Nothing Then Return
                End If
                Dim exeName As String = Path.GetFileName(Process.GetCurrentProcess().MainModule.FileName)
                _previousIeBrowserEmulationValue = registryKey.GetValue(exeName, 0)
                registryKey.SetValue(exeName, 11000, RegistryValueKind.DWord)
            Catch ex As Exception
                MessageCollector.AddExceptionMessage("IeBrowserEmulation.Register() failed.", ex, , True)
            End Try
        End Sub

        Public Shared Sub Unregister()
#If PORTABLE Then
            Try
                Dim registryKey As RegistryKey = Registry.CurrentUser.OpenSubKey(BrowserEmulationKey, True)
                If registryKey Is Nothing Then Return
                Dim exeName As String = Path.GetFileName(Process.GetCurrentProcess().MainModule.FileName)
                If _previousIeBrowserEmulationValue = 0 Then
                    registryKey.DeleteValue(exeName)
                Else
                    registryKey.SetValue(exeName, _previousIeBrowserEmulationValue, RegistryValueKind.DWord)
                End If
            Catch ex As Exception
                MessageCollector.AddExceptionMessage("IeBrowserEmulation.Unregister() failed.", ex, , True)
            End Try
#End If
        End Sub
    End Class
End Namespace

File: mRemoteV1.ConvertedToC#/Tools/IeBrowserEmulation.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.IO;
using Microsoft.Win32;
using mRemoteNG.App.Runtime;

namespace mRemoteNG.Tools
{
	public class IeBrowserEmulation
	{
		private const string BrowserEmulationKey = "Software\\Microsoft\\Internet Explorer\\Main\\FeatureControl\\FEATURE_BROWSER_EMULATION";
		private static int _previousIeBrowserEmulationValue = 0;
		public static void Register()
		{
			try {
				RegistryKey registryKey = Registry.CurrentUser.OpenSubKey(BrowserEmulationKey, true);
				if (registryKey == null) {
					Registry.CurrentUser.CreateSubKey(BrowserEmulationKey);
					registryKey = Registry.CurrentUser.OpenSubKey(BrowserEmulationKey, true);
					if (registryKey == null)
						return;
				}
				string exeName = Path.GetFileName(Process.GetCurrentProcess().MainModule.FileName);
				_previousIeBrowserEmulationValue = registryKey.GetValue(exeName, 0);
				registryKey.SetValue(exeName, 11000, RegistryValueKind.DWord);
			} catch (Exception ex) {
				mRemoteNG.App.Runtime.MessageCollector.AddExceptionMessage("IeBrowserEmulation.Register() failed.", ex, , true);
			}
		}

		#if PORTABLE
		public static void Unregister()
		{
			try {
				RegistryKey registryKey = Registry.CurrentUser.OpenSubKey(BrowserEmulationKey, true);
				if (registryKey == null)
					return;
				string exeName = Path.GetFileName(Process.GetCurrentProcess().MainModule.FileName);
				if (_previousIeBrowserEmulationValue == 0) {
					registryKey.DeleteValue(exeName);
				} else {
					registryKey.SetValue(exeName, _previousIeBrowserEmulationValue, RegistryValueKind.DWord);
				}
			} catch (Exception ex) {
				mRemoteNG.App.Runtime.MessageCollector.AddExceptionMessage("IeBrowserEmulation.Unregister() failed.", ex, , true);
			}
			#endif
		}
	}
}

